packages/client/hmi-client/src/components/model/tera-model-parts.vue

```diff
@@ -96,34 +96,22 @@ function updateMMT() {
 	});
 }
 
-function reset() {
+onMounted(() => {
 	transientModel.value = cloneDeep(props.model);
-}
-
-onMounted(() => updateMMT());
-
-// TODO: Do we still want autosave? It worked onUnmount but on staging the onbeforemount wasn't triggering
-// Apply changes to the model when the component unmounts or the user navigates away
-// onMounted(() => {
-// 	window.addEventListener('beforeunload', saveChanges);
-// 	updateMMT();
-// });
-
-// onUnmounted(() => {
-// 	saveChanges();
-// 	window.removeEventListener('beforeunload', saveChanges);
-// });
+	updateMMT();
+});
 
+// Only update the MMT when the semantics of the model changes outside this component.
 watch(
-	() => props.model,
-	() => {
-		reset();
-		updateMMT();
+	() => props.model.semantics,
+	(newSemantics) => {
+		if (!isEqual(newSemantics, transientModel.value.semantics)) {
+			transientModel.value = cloneDeep(props.model);
+			updateMMT();
+		}
```

Comment:
The properties in semantics aren't the only thing we are editing here. States and transitions are outside of semantics. So we may have to watch for those changes before updating the mmt